pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -58,3 +67,99 @@ func GetDeploymentMode(annotations map[string]string, deployConfig *v1beta1api.D
+func MergeRuntimeContainers(runtimeContainer *v1alpha1.Container, predictorContainer *v1.Container) (*v1.Container, error) {
+	// Default container configuration from the runtime.
+	coreContainer := v1.Container{
+		Args:            runtimeContainer.Args,
+		Command:         runtimeContainer.Command,
+		Env:             runtimeContainer.Env,
+		Image:           runtimeContainer.Image,
+		Name:            runtimeContainer.Name,
+		Resources:       runtimeContainer.Resources,
+		ImagePullPolicy: runtimeContainer.ImagePullPolicy,
+		WorkingDir:      runtimeContainer.WorkingDir,
+		LivenessProbe:   runtimeContainer.LivenessProbe,
+	}
+	// Save runtime container name, as the name can be overridden as empty string during the Unmarshal below
+	// since the Name field does not have the 'omitempty' struct tag.
+	runtimeContainerName := runtimeContainer.Name
+
+	// Args and Env will be combined instead of overridden.
+	argCopy := make([]string, len(coreContainer.Args))
+	copy(argCopy, coreContainer.Args)
+
+	envCopy := make([]v1.EnvVar, len(coreContainer.Env))
+	copy(envCopy, coreContainer.Env)
+
+	// Use JSON Marshal/Unmarshal to merge Container structs.
+	overrides, err := json.Marshal(predictorContainer)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(overrides, &coreContainer); err != nil {
+		return nil, err
+	}
+
+	if coreContainer.Name == "" {
+		coreContainer.Name = runtimeContainerName
+	}
+
+	argCopy = append(argCopy, predictorContainer.Args...)
+	envCopy = append(envCopy, predictorContainer.Env...)
+
+	coreContainer.Args = argCopy
+	coreContainer.Env = envCopy
+
+	return &coreContainer, nil
+
+}
+
+// Merge the predictor PodSpec struct with the runtime PodSpec struct, allowing users
+// to override runtime PodSpec settings from the predictor spec.
+func MergePodSpec(runtimePodSpec *v1alpha1.ServingRuntimePodSpec, predictorPodSpec *v1beta1.PodSpec) (*v1.PodSpec, error) {
+
+	corePodSpec := v1.PodSpec{
+		NodeSelector: runtimePodSpec.NodeSelector,
+		Affinity:     runtimePodSpec.Affinity,
+		Tolerations:  runtimePodSpec.Tolerations,
+	}
+
+	// Use JSON Marshal/Unmarshal to merge PodSpec structs.
+	overrides, err := json.Marshal(predictorPodSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := json.Unmarshal(overrides, &corePodSpec); err != nil {
+		return nil, err
+	}
+
+	return &corePodSpec, nil
+
+}
+
+// Get a ServingRuntime by name. First, ServingRuntimes in the given namespace will be checked.
+// If a resource of the specified name is not found, then ClusterServingRuntimes will be checked.
+func GetServingRuntime(cl client.Client, name string, namespace string) (*v1alpha1.ServingRuntimeSpec, error) {
+
+	runtime := &v1alpha1.ServingRuntime{}
+	err := cl.Get(context.TODO(), client.ObjectKey{Name: name, Namespace: namespace}, runtime)
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, err
+	} else if err == nil {
+		return &runtime.Spec, nil
```

Comment:
nit: could be simplified slightly (same below)
```suggestion
	if err == nil {
		return &runtime.Spec, nil
	} else if !errors.IsNotFound(err) {
		return &runtime.Spec, nil
	}
```

Reply:
Thanks, this simplifies it. Though, I am assuming you meant to have `return nil, err` in the last return,